core/src/main/java/bio/terra/pearl/core/service/kit/pepper/PepperApiException.java

```diff
@@ -0,0 +1,39 @@
+package bio.terra.pearl.core.service.kit.pepper;
+
+import lombok.Getter;
+import org.springframework.http.HttpStatus;
+
+/**
+ *  Class for 400-500 responses returned from Pepper.  This intentionally extends
```

Comment:
I think the error handling gets easier if we use this only at the lowest levels of the Pepper interface code (throw and catches), and convert the errors to something more generic that the rest of the stack can confidently process. Not wed to the concept, but I suggested having two uber exception classes like `JuniperBadRequest` and `JuniperInternalError` (or whatever naming works).  
This requires that the higher levels have no need to process a Pepper error any differently from the error from any service, but I suspect that is true for most/all service errors. (And in cases where it is not true, we might want to take a closer look at the cases.

Reply:
Interesting -- maybe we have those two base classes and then PepperApiException and friends extend them?  Then we don't have to do conversion -- each layer of the stack can operate on whatever granularity it likes?

Reply:
Sure, if that works for you. I suspect that additional shell is not necessary, but there is not a lot of downside to doing that way (unless the Spring exception handling stuff cannot handle the inheritance properly).